lib/galaxy/tools/verify/__init__.py

```diff
@@ -91,6 +91,8 @@ def verify(
         # if the server's env has GALAXY_TEST_SAVE, save the output file to that dir
         if keep_outputs_dir:
             ofn = os.path.join(keep_outputs_dir, filename)
+            if not os.path.isdir(ofn):
+                os.makedirs(os.path.dirname(ofn))
```

Comment:
`ofn` is a filename, so `os.path.isdir(ofn)` is always False, it should be:
```python
            # filename can contain a subdirectory part
            ofd = os.path.dirname(ofn)
            if not os.path.isdir(ofd):
                os.makedirs(ofd)
```

Reply:
oh .. sure .. my fault